src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -383,6 +356,17 @@ def create_virtual_machine(resource_group_name, vm_params, network_interfaces, a
         end
       end
 
+      unless vm_params[:config_disk].nil?
+        # append the config disk into the vm.
+        vm['properties']['storageProfile']['dataDisks'].push(
```

Comment:
if there is no ephemeral disk, an error will be raised in `vm['properties']['storageProfile']['dataDisks'].push`

Reply:
cool.